go/master/service.go

```diff
@@ -315,6 +328,25 @@ func (s *Service) logFields() log.Fields {
 	}
 }
 
+// updateQueuePasses check if tasks are done, move to todo to add another pass
+// IMPORTANT: must run under lock
+func (s *Service) updateQueuePasses() error {
```

Comment:
Look at the name from the English language perspective, "update pass" only mean "change pass", does not have explicit meaning of "moving to next pass", maybe `nextPass` could describe the intention better?